Stack_Config.h

```diff
@@ -0,0 +1,175 @@
+ *  STACK_ELEMENT_TYPE size-->variable that store the value of the size of the stack.
+ *  */
+typedef struct
+{
+	Node_t* pTop;
+	STACK_ELEMENT_TYPE size;
+}Stack_t;
+/*
+ *@brief: This function is used to creat the stack type.
+ *@Para : Stack_t* S::Pointer to the stack that will be created.
+ *@ret  : None.
+ *@Note : This function must be used with every linked stack.
+ *  */
+void LinkedStack_Creat(Stack_t* S);
+/*
+ *@brief: This function is used to push the data to the Linked Stack.
+ *@Para : Stack_t* S::Pointer to the stack that data will be pushed in.
+ *@Para : STACK_ELEMENT_TYPE Data::the data will be pushed in.
+ *@ret  : None.
+ *  */
+void LinkedStack_Push(Stack_t* S , STACK_ELEMENT_TYPE Data);
+/*
+ *@brief: This function is used to get the last element pushed in the stack.
+ *@Para : Stack_t* S::Pointer to the stack that you want to know the last element pushed in the stack .
+ *@Para : STACK_ELEMENT_TYPE *TopVal::the pointer will store the value of the last element pushed in the stack.
+ *@ret  : None.
+ *  */
+void LinkedStack_GetTop(Stack_t* S , STACK_ELEMENT_TYPE *TopVal);
+/*
+ *@brief: This function is used to pop  data from the Linked Stack.
+ *@Para : Stack_t* S::Pointer to the stack that data will be pulled out of it.
+ *@ret  : None.
+ *  */
+void LinkedStack_Pop(Stack_t* S);
+/*
+ *@brief: This function is used to check if stack is full or not.
+ *@Para : Stack_t* S::Pointer to the stack that you want to check if stack is full or not.
+ *@ret  : Return 0-->because the linked stack is always empty if there is place at memory.
+ *  */
+STACK_ELEMENT_TYPE LinkedStack_IsFull(Stack_t* S);
+/*
+ *@brief: This function is used to check if stack is empty or not.
+ *@Para : Stack_t* S::Pointer to the stack that you want to check if stack is empty or not.
+ *@ret  : Return 0-->if the stack is not empty.
+ *@ret  : Return 1-->if the stack is  empty.
+ *  */
+STACK_ELEMENT_TYPE LinkedStack_IsEmpty(Stack_t* S);
+/*
+ *@brief: This function is used to delete all the elements from the stack.
+ *@Para : Stack_t* S::Pointer to the stack that you want to clear it.
+ *@ret  : None.
+ *  */
+void LinkedStack_ClearStack(Stack_t* S);
+/*
+ *@brief: This function is used to traverse each element in the stack.
+ *@Para : Stack_t* S:pointer to the stack that you want to traverse it.
+ *@Para : void(*pFunction)(STACK_ELEMENT_TYPE Data):pFunction is a pointer to function which take one argument of the stack data type.
+ *@ret  : None.
+ *  */
+void LinkedStack_TraverseStack(Stack_t* S , void(*pFunction)(STACK_ELEMENT_TYPE Data));
+/*
+ *@brief: This function is used to get the size of the stack.
+ *@Para : Stack_t* S::Pointer to the stack that you want to know the size of it.
+ *@Para : STACK_ELEMENT_TYPE *Size::the pointer will store the value of the size of the stack.
+ *@ret  : None.
+ *  */
+void LinkedStack_GetSize(Stack_t* S , STACK_ELEMENT_TYPE *Size);
+
+#endif /* STACK_CONFIG_H_ */
```

Comment:
This configuration file have to contain only all options that will be changed from the user for example : 
`#define STACK_TYPE       STACK_ARRAY`    
This file is normal header file and must be named `Stack.h`